### ISSUE TITLE
Do report deduplication after an Analysis run like what is done during the Store command

### ISSUE THREAD

**Issue by vodorok** [Ericsson/codechecker#4042]
Currently, when one tries to store a report folder, and there are multiple same reports from a header file, only one piece report will be kept and stored of the identical reports.

This can be extremely wasteful on large report folders when the raw report count is in the millions. It is also confusing when the store command uploads the report folder, it will print the raw report count which will be different to the report count that is actually stored. 

See the following examples:
CodeChecker parse <results_folder>
```
----==== Checker Statistics ====----                                      
-------------------------------------------------------------------------                                                                           
Checker name                               | Severity | Number of reports                                                                           
------------------------------------------------------------------------- 
cppcheck-uninitMemberVar                   | MEDIUM   |                 4
cppcheck-invalidPrintfArgType_uint         | MEDIUM   |                 1
cppcheck-invalidPrintfArgType_sint         | MEDIUM   |                 2
cppcoreguidelines-virtual-class-destructor | MEDIUM   |                12
cppcoreguidelines-special-member-functions | LOW      |                17
bugprone-sizeof-expression                 | HIGH     |                 1 
-------------------------------------------------------------------------
----=================----            
                                     
----==== File Statistics ====----    
--------------------------------     
File name    | Number of reports                                          
--------------------------------                                          
tinyxml2.h   |                36                                          
tinyxml2.cpp |                 1                                          
--------------------------------                                          
----=================----      

```
Store result on GUI:
![image](https://github.com/Ericsson/codechecker/assets/23135800/09bdb3e4-910c-44bf-a041-3e23d05b1b23)

CodeChecker store <results_folder>
```
----==== Checker Statistics ====----                                                                                                                
-------------------------------------------------------------------------                                                                           
Checker name                               | Severity | Number of reports                                                                           
-------------------------------------------------------------------------                                                                           
cppcheck-uninitMemberVar                   | MEDIUM   |                 7                                                                           
cppcheck-invalidPrintfArgType_uint         | MEDIUM   |                 1                                                                           
cppcheck-invalidPrintfArgType_sint         | MEDIUM   |                 4
cppcoreguidelines-virtual-class-destructor | MEDIUM   |                24
cppcoreguidelines-special-member-functions | LOW      |                34                                                                           
bugprone-sizeof-expression                 | HIGH     |                 1
-------------------------------------------------------------------------
----=================----

----==== File Statistics ====----
--------------------------------
File name    | Number of reports
--------------------------------
tinyxml2.h   |                70
tinyxml2.cpp |                 1
--------------------------------
----=================----
```

It would be a beneficial feature to deduplicate identical reports during or after the analysis run. The same algorithm should be used that the store handler uses, to ensure the best compatibility.

**Comment by bruntib** [comment]
@vodorok Could you, please, check if this is still a problem? I can't reproduce these numbers. Could have it been fixed? Thank you!
